codespeed/views.py

```diff
@@ -731,3 +739,239 @@ def add_json_results(request):
     logger.debug("add_json_results: completed")
 
     return HttpResponse("All result data saved successfully", status=202)
+
+
+def django_has_content_type():
+    return (django.VERSION[0] > 1 or
+            (django.VERSION[0] == 1 and django.VERSION[1] >= 6))
+
+
+def validate_results_request(data):
```

Comment:
Good catch! `validate_result` has a bad interface. It returns `Environment, error`, so it really does two things: validate the input and fetch an environment. That tells us it should be split.

Anyway your suggestion makes total sense